pkg/apis/core/v1/defaults.go

```diff
@@ -429,3 +435,56 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+// defaultPodRequests applies default values for pod-level requests, only when
+// pod-level limits are set, in following scenarios:
+// 1. When at least one container (regular, init or sidecar) has requests set:
+// The pod-level requests become equal to the effective requests of all containers
+// in the pod.
+// 2. When no containers have requests set: The pod-level requests become equal to
+// pod-level limits.
+// This defaulting behavior ensures consistent resource accounting at the pod-level
+// while maintaining compatibility with the container-level specifications, as detailed
+// in KEP-2837: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2837-pod-level-resource-spec/README.md#proposed-validation--defaulting-rules
+func defaultPodRequests(obj *v1.Pod) {
+	// We only populate defaults when the pod-level resources are partly specified already.
+	if obj.Spec.Resources == nil {
+		return
+	}
+
+	if len(obj.Spec.Resources.Limits) == 0 {
+		return
+	}
+
+	var reqs v1.ResourceList
+	reqs = obj.Spec.Resources.Requests
```

Comment:
super nitpick, but since this code is SO important and is subtle:

1) Consider renaming `reqs` to `podReqs`
2) Consider renaming `podReqsFromContainers` to `aggCtrReqs` or `effPodReqs` or `sumCtrReqs`
3) Consider renaming `value` to `podLim` or `lim`

Thus, things like `reqs[key] = value.DeepCopy()` -> `podReqs[key] = podLim.DeepCopy()`